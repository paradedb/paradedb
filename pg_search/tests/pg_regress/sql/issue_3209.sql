\i common/common_setup.sql

DROP TABLE IF EXISTS filing;
CREATE TABLE filing (id serial, form text, filing TEXT,constraint filing_pkey primary key (id, form)) partition by list (form);
CREATE INDEX filing_idx ON filing USING bm25 (id, form, filing) with (key_field = 'id', text_fields = '{"form": {"fast": true, "tokenizer": {"type": "keyword"}}}');
CREATE TABLE filing_10_k PARTITION OF filing FOR VALUES IN ('10-K', '10-K/A');
INSERT INTO filing (form, filing) VALUES ('10-K', 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.');
INSERT INTO filing (form, filing) VALUES ('10-K/A', 'Sed do eiusmod tempor incididunt ut labore et dolore magna aliqua');
SELECT paradedb.snippet(filing) FROM filing WHERE filing @@@ 'lorem';
DROP TABLE filing;
