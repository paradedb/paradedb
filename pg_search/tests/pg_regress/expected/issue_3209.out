\i common/common_setup.sql
CREATE EXTENSION IF NOT EXISTS pg_search;
-- Disable parallel workers to avoid differences in plans
SET max_parallel_workers_per_gather = 0;
SET enable_indexscan to OFF;
SET paradedb.enable_mixed_fast_field_exec = true;
DROP TABLE IF EXISTS filing;
CREATE TABLE filing (id serial, form text, filing TEXT,constraint filing_pkey primary key (id, form)) partition by list (form);
CREATE INDEX filing_idx ON filing USING bm25 (id, form, filing) with (key_field = 'id', text_fields = '{"form": {"fast": true, "tokenizer": {"type": "keyword"}}}');
CREATE TABLE filing_10_k PARTITION OF filing FOR VALUES IN ('10-K', '10-K/A');
INSERT INTO filing (form, filing) VALUES ('10-K', 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.');
INSERT INTO filing (form, filing) VALUES ('10-K/A', 'Sed do eiusmod tempor incididunt ut labore et dolore magna aliqua');
SELECT paradedb.snippet(filing) FROM filing WHERE filing @@@ 'lorem';
                            snippet                             
----------------------------------------------------------------
 <b>Lorem</b> ipsum dolor sit amet, consectetur adipiscing elit
(1 row)

DROP TABLE filing;
